### PR TITLE
Fix bug in displaying name of spell after winning pyramid

### PR DESCRIPTION
**Pull request by jkoppel**

### CODE REVIEW

**Comment by jkoppel** [comment]
@XT3RM1N8R This extends some of your code. Please review.

File: src/cpp/shared/adventure/adv.cpp [hunk 1]
```diff
@@ -133,11 +133,15 @@ void advManager::DoEvent(class mapCell *cell, int locX, int locY) {
   SAMPLE2 res2 = NULL_SAMPLE2;
   nonstd::optional<bool> shouldSkip = ScriptCallbackResult<bool>("OnLocationVisit", locationType, locX, locY);
   if (!shouldSkip.value_or(false)) {
-    if (locationType != LOCATION_SHRINE_FIRST_ORDER && locationType != LOCATION_SHRINE_SECOND_ORDER && locationType != LOCATION_SHRINE_THIRD_ORDER) {
+
```
Comment (Kert) [inline src/cpp/shared/adventure/adv.cpp]: Useless new line?

File: src/cpp/shared/adventure/adv.cpp [hunk 2]
```diff
@@ -133,11 +133,15 @@ void advManager::DoEvent(class mapCell *cell, int locX, int locY) {
...
+    if (locationType == LOCATION_SHRINE_FIRST_ORDER || locationType == LOCATION_SHRINE_SECOND_ORDER || locationType == LOCATION_SHRINE_THIRD_ORDER) {
```
Comment (Kert) [inline src/cpp/shared/adventure/adv.cpp]: Maybe using switch-case instead? Since there are 3 options here

File: src/cpp/shared/adventure/adv.cpp [hunk 3]
```diff
@@ -189,6 +193,53 @@ void advManager::HandleSpellShrine(class mapCell *cell, int locationType, hero *
...
+void advManager::HandlePyramid(class mapCell *cell,int locType, hero *hro, SAMPLE2 *res2, int locX, int locY) {
+  this->EventSound(locType, cell->extraInfo, res2);
+
+  this->EventWindow(-1, 2,
+    "You come upon the pyramid of a great and ancient king.  You are tempted to search it for treasure, but all the old stories warn of fearful curses and undead guardians.  Will you search?",
+    -1, 0, -1, 0, -1);
+
+  if (gpWindowManager->buttonPressedCode == BUTTON_YES) {
+    if (cell->extraInfo != 0) {
+
```
Comment (Kert) [inline src/cpp/shared/adventure/adv.cpp]: Useless new line?

File: src/cpp/shared/adventure/adv.cpp [hunk 4]
```diff
@@ -189,6 +193,53 @@ void advManager::HandleSpellShrine(class mapCell *cell, int locationType, hero *
...
+  if (gpWindowManager->buttonPressedCode == BUTTON_YES) {
+    if (cell->extraInfo != 0) {
+      
+      if (!this->CombatMonsterEvent(hro, CREATURE_ROYAL_MUMMY, 30, cell, locX, locY, 0, locX, locY, CREATURE_VAMPIRE_LORD, 20, 2, -1, 0, 0)) {
+        hro->CheckLevel();
+        
+        std::string msg;
+        msg += "Upon defeating the monsters, you decipher an ancient glyph on the wall, telling the secret of the spell - '";
+        msg += gSpellNames[cell->extraInfo - 1];
```
Comment (Kert) [inline src/cpp/shared/adventure/adv.cpp]: Make a variable out of cell->extraInfo - 1?

File: src/cpp/shared/adventure/adv.cpp (hunk 2 above)
Comment (jkoppel) [reply src/cpp/shared/adventure/adv.cpp]: Good idea.

**Pull request closed without merging**